edb/pgsql/compiler/relctx.py

```diff
@@ -1740,34 +1739,6 @@ def range_for_material_objtype(
                 rvar = rvar_for_rel(
                     sctx.rel, lateral=lateral, typeref=typeref, ctx=sctx)
 
-    else:
-        assert not typeref.is_view, "attempting to generate range from view"
```

Comment:
Could we preserve this assert in the new path?

Reply:
I put it in `_table_from_typeref` but I think it's reasonable to have it more ahead in the process.